watertap/examples/flowsheets/case_studies/BSM2/ASM_ADM_integration.py

```diff
@@ -0,0 +1,422 @@
+        initialize=5.7,
+        units=pyo.units.hour**-1,
+        doc="Lumped mass transfer coefficient for oxygen",
+    )
+    m.fs.S_O_eq = pyo.Param(
+        default=8e-3,
+        units=pyo.units.kg / pyo.units.m**3,
+        mutable=True,
+        doc="Dissolved oxygen concentration at equilibrium",
+    )
+
+    @m.fs.R3.Constraint(m.fs.time, doc="Mass transfer constraint for R3")
+    def mass_transfer_R3(self, t):
+        return pyo.units.convert(
+            m.fs.R3.injection[t, "Liq", "S_O"], to_units=pyo.units.kg / pyo.units.hour
+        ) == (
+            m.fs.R3.KLa
+            * m.fs.R3.volume[t]
+            * (m.fs.S_O_eq - m.fs.R3.outlet.conc_mass_comp[t, "S_O"])
+        )
+
+    @m.fs.R4.Constraint(m.fs.time, doc="Mass transfer constraint for R4")
+    def mass_transfer_R4(self, t):
+        return pyo.units.convert(
+            m.fs.R4.injection[t, "Liq", "S_O"], to_units=pyo.units.kg / pyo.units.hour
+        ) == (
+            m.fs.R4.KLa
+            * m.fs.R4.volume[t]
+            * (m.fs.S_O_eq - m.fs.R4.outlet.conc_mass_comp[t, "S_O"])
+        )
+
+    # Feed Water Conditions
+    m.fs.FeedWater.flow_vol.fix(20648 * pyo.units.m**3 / pyo.units.day)
+    m.fs.FeedWater.temperature.fix(308.15 * pyo.units.K)
+    m.fs.FeedWater.pressure.fix(1 * pyo.units.atm)
+    m.fs.FeedWater.conc_mass_comp[0, "S_I"].fix(27 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "S_S"].fix(58 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_I"].fix(92 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_S"].fix(363 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_BH"].fix(50 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_BA"].fix(0 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_P"].fix(0 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "S_O"].fix(0 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "S_NO"].fix(0 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "S_NH"].fix(23 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "S_ND"].fix(5 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.conc_mass_comp[0, "X_ND"].fix(16 * pyo.units.g / pyo.units.m**3)
+    m.fs.FeedWater.alkalinity.fix(7 * pyo.units.mol / pyo.units.m**3)
+
+    # Reactor sizing
+    m.fs.R1.volume.fix(1000 * pyo.units.m**3)
+    m.fs.R2.volume.fix(1000 * pyo.units.m**3)
+    m.fs.R3.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R4.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R5.volume.fix(1333 * pyo.units.m**3)
+
+    # Injection rates to Reactions 3, 4 and 5
+    for j in m.fs.props_ASM1.component_list:
+        if j != "S_O":
+            # All components except S_O have no injection
+            m.fs.R3.injection[:, :, j].fix(0)
+            m.fs.R4.injection[:, :, j].fix(0)
+            m.fs.R5.injection[:, :, j].fix(0)
```

Comment:
Should these also be incorporated into the respective  CSTR_injection unit model?